### PR TITLE
Add Trip Functionality and Updated User Authentication Verification

### PR DESCRIPTION
**Pull request by nowshin-108**
# Test Plan for Add Trip Functionality and User Authentication Verification

## 1. Introduction
This document outlines the test strategy and overall framework for conducting functional testing on the "Add Trip" feature and the updated user authentication verification process.

## 2. Test Items
- Add Trip functionality
- User authentication verification using session-based authentication
- Amadeus API integration for flight search
- Frontend display of flight search results
- Backend session management and API route protection
- Loading state and single error state error handling

## 3. Features to be Tested
- Correct implementation of flight search using the Amadeus API.
- Accurate display of flight search results in the user interface.
- Ability to add a trip with the selected flight details to the database.
- Proper functioning of session-based user authentication.
- Protection of API routes through middleware that verifies user sessions.

## 4. Video Walkthrough
https://www.loom.com/share/1c1529dccc444a1bb68773c2e7972b23?sid=6c279191-5344-4fb1-adc7-69fc052cf1b8

### CODE REVIEW

File: backend/routes/flightStatus.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+const prisma = new PrismaClient();
+const router = express.Router();
+const amadeus = new Amadeus({
+    clientId: API_KEY,
+    clientSecret: API_SECRET,
+});
+
+//flight search for adding trip - needs authentication
+router.get('/flight-status', authenticateUser, async (req, res) => {
```
Comment (alantulais) [inline backend/routes/flightStatus.js:15]: So this "authenticateUser" is a middleware, right?

File: backend/routes/flightStatus.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+});
+
+
+// saving static trip details on adding trip - needs authentication
+router.post('/add-trip', authenticateUser, async (req, res) => { 
+    try { 
+        const { carrierCode, flightNumber, scheduledDepartureDate } = req.body; 
+        const userId = req.session.user.userId;
+        const newTrip = await prisma.trip.create({ 
+        data: { userId, carrierCode, flightNumber, scheduledDepartureDate }
```
Comment (alantulais) [inline backend/routes/flightStatus.js:37]: So you won't store any data from the external API locally?

File: frontend/src/App.jsx [hunk 3]
```diff
@@ -1,37 +1,64 @@
...
 
   const updateUser = (newUser) => {
     setUser(newUser);
   };
 
   useEffect(() => {
-    localStorage.setItem('user', JSON.stringify(user));
-  }, [user]);
+    //making a call to backend middleware API to check the authentication status of user
+    const checkAuth = async () => {
```
Comment (alantulais) [inline frontend/src/App.jsx:25]: Actually, is it possible to perform this authentication in the express router directly? I might have missed that, you can do that directly in express before rendering routes that are supposed to be protected.

File: frontend/src/App.jsx [hunk 4]
```diff
@@ -1,37 +1,64 @@
...
+      try {
+        const response = await fetch(`${API_BASE_URL}/users/auth-check`, {
+          credentials: 'include'
+        });
+        if (response.ok) {
+          const data = await response.json();
+          setUser(data.user);
+        }
+      } catch (error) {
+        console.error('Auth check failed:', error);
```
Comment (alantulais) [inline frontend/src/App.jsx:35]: You might need to redirect to login if this fails

File: frontend/src/Components/HomePage/AddTrip/FlightForm.jsx [hunk 5]
```diff
@@ -0,0 +1,242 @@
...
+    const [error, setError] = useState(null);
+    const [isLoading, setIsLoading] = useState(false);
+    const [flightAdded, setFlightAdded] = useState(() => sessionStorage.getItem('flightAdded') === 'true');
+    const [isAddingFlight, setIsAddingFlight] = useState(false);
+
+
+
+    useEffect(() => {
+
+        sessionStorage.setItem('flightNumber', flightNumber);
```
Comment (alantulais) [inline frontend/src/Components/HomePage/AddTrip/FlightForm.jsx:31]: Let's try to use context as suggested by Abe.

**Review by alantulais — APPROVED** [review]

File: backend/routes/flightStatus.js (hunk 1 above)
Comment (nowshin-108) [reply backend/routes/flightStatus.js:15]: Yeap. I have that function in auth.js and importing it where needed.

File: backend/routes/flightStatus.js (hunk 2 above)
Comment (nowshin-108) [reply backend/routes/flightStatus.js:37]: Right now, I'm only saving the essentials like carrierCode, flightNumber, scheduledDeparture, and userID. This helps us quickly fetch flight statuses for the upcoming trips page without storing too much data upfront. Later I'll use it to fetch realtime data for upcoming pages.
Once the flight lands, we'll save the full details in a new table, maybe called PastTrips. This way, we can easily show all the past flights on the Past Trip page.
Let me know if you think we should tweak this approach or if there's anything else to consider!

File: frontend/src/App.jsx (hunk 3 above)
Comment (nowshin-108) [reply frontend/src/App.jsx:25]: I'll look into it and let you know any updates

File: frontend/src/App.jsx (hunk 4 above)
Comment (nowshin-108) [reply frontend/src/App.jsx:35]: Gotcha! Thanks for catching it

File: frontend/src/Components/HomePage/AddTrip/FlightForm.jsx (hunk 5 above)
Comment (nowshin-108) [reply frontend/src/Components/HomePage/AddTrip/FlightForm.jsx:31]: Will do!

**Pull request merged**